talk/concurrency.tex

```diff
@@ -233,13 +233,43 @@
       int a = 0;
       std::mutex m;
       void inc() {
-        std::lock_guard<std::mutex> guard(m);
+        std::scoped_lock lock{m};
         a++;
       }
     \end{cppcode*}
   \end{exampleblock}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitleii{Mutexes and Locks}
+  \begin{block}{Good practice}
+    \begin{itemize}
+      \item Always use \texttt{scoped\_lock} (\cpp11: \texttt{lock\_guard})
+      \item Hold as short as possible, wrap critical section in "\texttt{\string{  \string}}"
+      \item Only if manual control needed, use \texttt{unique\_lock}
+    \end{itemize}
+  \end{block}
+  \begin{exampleblock}{}
+    \begin{cppcode*}{gobble=2}
+      {
+        std::scoped_lock myLocks{mutex1, mutex2, ...};
+        // critical section
+      }
+    \end{cppcode*}
+  \end{exampleblock}{}
+  \pause
+  \begin{exampleblock}{}
+    \begin{cppcode*}{gobble=2}
+      std::unique_lock lock1{mutex1, std::try_to_lock};
+      std::unique_lock lock2{mutex2, std::defer_lock};
+      if (lock1) {
+        std::lock(lock2);
+        // ...
+      }
+    \end{cppcode*}
+  \end{exampleblock}
```

Comment:
Does this example really help? I know `try_to_lock` and `defer_lock` exist, but how often do you really need them?

Maybe use this additional space to mention `shared_lock`.

Reply:
Also pushed to #62. I'll leave this space blank in the simple version.